test/unit/spec/appWidgets/input/checkboxInputSpec.js

```diff
@@ -169,18 +178,24 @@ define(['widgets/appWidgets2/input/checkboxInput', 'common/runtime', 'testUtil']
                 ).not.toBeNull();
                 // it should have a message
                 expect(errorContainer.textContent).toContain(
-                    `Invalid value of "${testConfig.initialValue}" for parameter ${testConfig.parameterSpec.ui.label}. Default value unchecked used.`
+                    `Invalid value of "${testConfig.initialValue}" for parameter ${testConfig.parameterSpec.ui.label}. Default value of unchecked used.`
                 );
             });
 
-            it('the error should be dismissable with a button', async () => {
+            it('the error should be dismissable with a button, and propagate the default value', async () => {
                 const widget = CheckboxInput.make(testConfig);
+                let validMsg;
+                bus.on('changed', (msg) => {
+                    validMsg = msg;
```

Comment:
you could put the test (`expect(msg).toEqual({ newValue: ... })`) here instead of using `TestUtil.wait()` and then checking the value

Reply:
There's no real guarantee that'll get called before the test expects to finish. Though it could return as a Promise. That might be better.